cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -72,33 +75,37 @@ private URI addCookieInterceptorIfRequired(URI uri) {
                     "Protocol %s not supported", uriProtocol));
         }
 
-        if (uri.getUserInfo() != null) {
+        if (uri.getUserInfo() != null && this.username == null && this.password == null) {
             String[] parts = uri.getUserInfo().split(":");
-            if (parts.length == 2) {
+            this.username = parts[0];
```

Comment:
Should probably still have the `if (parts.length == 2)` check - otherwise this will throw OOB exception for a URL of the format `http://user@host/abc`